pybricks/media/pb_type_image.c

```diff
@@ -0,0 +1,353 @@
+
+    pb_assert_type(source_in, &pb_type_Image);
+    pb_type_Image_obj_t *source = MP_OBJ_TO_PTR(source_in);
+
+    int x = (self->image.width - source->image.width) / 2;
+    int y = (self->image.height - source->image.height) / 2;
+
+    pbio_image_fill(&self->image, 0);
+    pbio_image_draw_image(&self->image, &source->image, x, y);
+
+    if (self->is_display) {
+        pbdrv_display_update();
+    }
+
+    return mp_const_none;
+}
+static MP_DEFINE_CONST_FUN_OBJ_KW(pb_type_Image_load_image_obj, 1, pb_type_Image_load_image);
+
+static mp_obj_t pb_type_Image_draw_image(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    PB_PARSE_ARGS_METHOD(n_args, pos_args, kw_args,
+        pb_type_Image_obj_t, self,
+        PB_ARG_REQUIRED(x),
+        PB_ARG_REQUIRED(y),
+        PB_ARG_REQUIRED(source),
+        PB_ARG_DEFAULT_NONE(transparent));
+
+    mp_int_t x = mp_obj_get_int(x_in);
+    mp_int_t y = mp_obj_get_int(y_in);
+    pb_assert_type(source_in, &pb_type_Image);
+    pb_type_Image_obj_t *source = MP_OBJ_TO_PTR(source_in);
+    bool transparent = false;
+    int transparent_value;
+    if (transparent_in != mp_const_none) {
+        transparent = true;
+        transparent_value = get_color(transparent_in);
+    }
+
+    if (!transparent) {
+        pbio_image_draw_image(&self->image, &source->image, x, y);
+    } else {
+        pbio_image_draw_image_transparent(&self->image, &source->image, x, y, transparent_value);
+    }
+
+    if (self->is_display) {
+        pbdrv_display_update();
+    }
+
+    return mp_const_none;
+}
+static MP_DEFINE_CONST_FUN_OBJ_KW(pb_type_Image_draw_image_obj, 1, pb_type_Image_draw_image);
+
+static mp_obj_t pb_type_Image_draw_pixel(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    PB_PARSE_ARGS_METHOD(n_args, pos_args, kw_args,
+        pb_type_Image_obj_t, self,
+        PB_ARG_REQUIRED(x),
+        PB_ARG_REQUIRED(y),
+        PB_ARG_DEFAULT_NONE(color));
+
+    mp_int_t x = pb_obj_get_int(x_in);
+    mp_int_t y = pb_obj_get_int(y_in);
+    int color = get_color(color_in);
+
+    pbio_image_draw_pixel(&self->image, x, y, color);
+
+    if (self->is_display) {
+        pbdrv_display_update();
+    }
+
+    return mp_const_none;
+}
+static MP_DEFINE_CONST_FUN_OBJ_KW(pb_type_Image_draw_pixel_obj, 1, pb_type_Image_draw_pixel);
+
+static mp_obj_t pb_type_Image_draw_line(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    PB_PARSE_ARGS_METHOD(n_args, pos_args, kw_args,
+        pb_type_Image_obj_t, self,
+        PB_ARG_REQUIRED(x1),
+        PB_ARG_REQUIRED(y1),
+        PB_ARG_REQUIRED(x2),
+        PB_ARG_REQUIRED(y2),
+        PB_ARG_DEFAULT_INT(width, 1),
+        PB_ARG_DEFAULT_NONE(color));
+
+    mp_int_t x1 = pb_obj_get_int(x1_in);
+    mp_int_t y1 = pb_obj_get_int(y1_in);
+    mp_int_t x2 = pb_obj_get_int(x2_in);
+    mp_int_t y2 = pb_obj_get_int(y2_in);
+    mp_int_t width = pb_obj_get_int(width_in);
+    int color = get_color(color_in);
+
```

Comment:
Might want to `mp_raise_NotImplementedError()` here since the pbio functions silently don't do anything currently. Unless we intend to implement those right away. (Bert, likes to test everything right away, so he might report a bug :grin:)